File: 102.cpp
#include "Tree.h"
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<TreeNode*> q;
        vector<vector<int>> res;
        if (root)
            q.push(root);

        while (!q.empty()) {
            int size = q.size();
            vector<int> row;

            while (size--) {
                TreeNode* temp = q.front();
                q.pop();
                row.push_back(temp->val);
                if (temp->left)
                    q.push(temp->left);
                if (temp->right)
                    q.push(temp->right);
            }

            res.push_back(row);
        }

        return res;
    }
};

int main() {
    TreeEnv t;
    vector<int> list = {3, 9, 20, NULL_TREENODE, NULL_TREENODE, 15, 7};
    Solution s;

    TreeNode* root = t.initialTree(list);
    vector<vector<int>> res = s.levelOrder(root);

    for (auto row : res) {
        cout << "[";
        for (auto node : row)
            cout << node << ", ";
        cout << "]" << endl;
    }
    cout << endl;

    return 0;
}

File: Python/102.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        d = deque()
        if root:
            d.append(root)
        res = []
        while len(d) != 0:
            n = len(d)
            l = []
            for i in range(n):
                node = d.popleft()
                l.append(node.val)
                if node.left:
                    d.append(node.left)
                if node.right:
                    d.append(node.right)
            res.append(l)
        return res
        
        
